@xen-orchestra/build-template/src/index.spec.js

```diff
@@ -0,0 +1,14 @@
+/* eslint-env jest */
+import buildTemplate from '.'
+
+it("correctly replaces the template's variables", () => {
+  const replacer = buildTemplate(
+    '{property}_\\{property}_\\\\{property}_{constant}_%_FOO',
+    {
+      '{property}': obj => obj.name,
+      '{constant}': 1235,
+      '%': (_, i) => i,
+    }
+  )
+  expect(replacer({ name: 'bar' }, 5)).toBe('bar_{property}_\\bar_1235_5_FOO')
+})
```

Comment:
Simplify your test, you need to test replacing:
- a constant
- the first arg
- the second arg

Reply:
Can you please give me a example?

Reply:
Never mind, this is unimportant.